src/cc_catalog_airflow/dags/provider_api_scripts/cleveland_museum_of_art.py

```diff
@@ -0,0 +1,213 @@
+            logger.info(f'Total Images till now {images_till_now}')
+            offset = offset + LIMIT
+        else:
+            logger.info(f'No more images to process so exiting the loop')
+            condition = False
+    total_images = image_store.commit()
+    logger.info(f'Total number of images received {total_images}')
+
+
+
+def _build_query_params(
+        start_date,
+        end_date,
+        default_query_params=DEFAULT_QUERY_PARAMS,
+):
+    query_params = default_query_params.copy()
+    query_params['gaistart'] = start_date
+    query_params['gaiend'] = end_date
+    return query_params
+
+
+def _derive_timestamp_pair(date):
+    date_obj = datetime.strptime(date, '%Y-%m-%d')
+    utc_date = date_obj.replace(tzinfo=timezone.utc)
+    start_timestamp = str(int(utc_date.timestamp()))
+    end_timestamp = str(int((utc_date + timedelta(days=1)).timestamp()))
+    return start_timestamp, end_timestamp
+
+
+def _get_response_json(
+        query_params,
+        endpoint,
+        request_headers=DEFAULT_REQUEST_HEADERS,
+        retries=5,
+):
+    response_json = None
+
+    if retries < 0:
+        logger.error('No retries remaining.  Failure.')
+        raise Exception('Retries exceeded')
+
+    response = delayed_requester.get(
+        endpoint,
+        params=query_params,
+        headers=request_headers,
+        timeout=60
+    )
+    if response is not None and response.status_code == 200:
+        try:
+            response_json = response.json()
+        except Exception as e:
+            logger.warning(f'Could not get response_json.\n{e}')
+            response_json = None
+
+    if (
+            response_json is None
+            or response_json.get('error') is not None
+    ):
+        logger.warning(f'Bad response_json:  {response_json}')
+        logger.warning(
+            'Retrying:\n_get_response_json(\n'
+            f'    {endpoint},\n'
+            f'    {query_params},\n'
+            f'    {request_headers}'
+            f'    retries={retries - 1}'
+            ')'
+        )
+        response_json = _get_response_json(
+            query_params,
+            endpoint=endpoint,
+            request_headers=request_headers,
+            retries=retries - 1
+        )
+
+    return response_json
+
+
+
+
+def handle_the_response(response):
+    for data in response:
+        foreign_url = data.get('url', None)
+        single_image = data.get('images')
+        image_url = None
+        images_till_now = 0
+        if single_image is not None:
+            if single_image.get('web'):
+                key = 'web'
+                image_url = single_image.get('web').get('url', None)
+            elif single_image.get('print'):
+                key = 'print'
+                image_url = single_image.get('print').get('url', None)
+            else:
+                key = 'full'
+                image_url = single_image.get('full').get('url', None)
```

Comment:
Note, the lines
```python
else:
    key = 'full'
    image_url = single_image.get('full').get('url', None)
```
will raise an exception in the case we reach this branch, but there is no key `full`.  This would be a problem

Reply:
So what should be the expected behaviour if `full` isn't found ? `return None` ?